resilience4j-retry/src/main/java/io/github/resilience4j/retry/Retry.java

```diff
@@ -510,9 +510,16 @@ default void executeRunnable(Runnable runnable) {
 
 		/**
 		 * Records a successful call.
+		 * @deprecated since 1.2.0
 		 */
+		@Deprecated
 		void onSuccess();
 
+		/**
+		 * Records a successful call or retryable call with the needed generated retry events.
```

Comment:
Could you please enhance this JavaDoc and describe what happens on onComplete to the user.